src/recoveryUtils.go

```diff
@@ -0,0 +1,484 @@
+			ersDecryptor = ers.NewRSADecryptor(key)
+		} else if recoveryMethod == HSM_TOKEN {
+			ersHsmHelper = InitializeErsHsmHelper()
+		} else if recoveryMethod == MOBILE_PRIVATE_KEY {
+			log.Fatal("Mobile private key recovery method is not supported for server backup recovery")
+		}
+	} else if recoveryType == MOBILE {
+		backupFileName := TakeInput("Enter recovery package file name")
+		recoveryInfo = getRecoveryInfoFromMobileBackup(backupFileName, algorithm)
+
+		if recoveryMethod == LOCAL_PRIVATE_KEY {
+			password := TakePasswordInput("WARNING: PERFORM THIS ACTION ONLY ON OFFLINE COMPUTER\n" +
+				"Enter Recovery key pair passphrase")
+			key = readPrivateKeyFromPemFile(password)
+			ersDecryptor = ers.NewRSADecryptor(key)
+		} else if recoveryMethod == HSM_TOKEN {
+			ersHsmHelper = InitializeErsHsmHelper()
+		} else if recoveryMethod == MOBILE_PRIVATE_KEY {
+			password := TakePasswordInput("WARNING: PERFORM THIS ACTION ONLY ON OFFLINE COMPUTER\n" +
+				"Enter Recovery key pair passphrase")
+			key = getPrivateKeyFromMobileBackup(backupFileName, password)
+			ersDecryptor = ers.NewRSADecryptor(key)
+		}
+	}
+
+	fmt.Println("\nProcessing recovery data, please wait for a response.")
+
+	return recoveryMethod, recoveryInfo, &key.PublicKey, ersHsmHelper, ersDecryptor
+}
+
+func handleErsRecoverPrivateKey(recoveryMethod RecoveryMethod, ersHsmHelper *ErsHsmHelper, ersDecryptor ers.Decryptor, recoveryData []byte, chainPath []uint32) (string, []byte, []byte) {
+	var ellipticCurve string
+	var privateKeyASN1 []byte
+	var masterChainCode []byte
+	var err error
+
+	if recoveryMethod == HSM_TOKEN && ersHsmHelper != nil {
+		ellipticCurve, privateKeyASN1, masterChainCode, err = ers.RecoverPrivateKey(ersHsmHelper, []byte(""), recoveryData, chainPath)
+	} else if ersDecryptor != nil {
+		ellipticCurve, privateKeyASN1, masterChainCode, err = ers.RecoverPrivateKey(ersDecryptor, []byte(""), recoveryData, chainPath)
+	} else {
+		log.Fatal("ERS decryptor not found for recovery method")
+	}
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return ellipticCurve, privateKeyASN1, masterChainCode
+}
+
+func handleEncryptRecoveredData(recoveryMethod RecoveryMethod, ersHsmHelper *ErsHsmHelper, rsaPublicKey *rsa.PublicKey, plainText []byte, label []byte) []byte {
+	var encryptedBytes []byte
+	var err error
+
+	if recoveryMethod == HSM_TOKEN {
+		encryptedBytes, err = ersHsmHelper.Encrypt(plainText, label)
+	} else {
+		if rsaPublicKey == nil {
+			log.Fatal("Invalid private key")
+		}
+
+		encryptedBytes, err = rsa.EncryptOAEP(
+			sha256.New(),
+			rand.Reader,
+			rsaPublicKey,
+			plainText,
+			label,
+		)
+	}
+
+	checkError(err, "Unable to encrypt recovered data")
+
+	return encryptedBytes
+}
+
+func readPrivateKeyFromPemFile(password string) *rsa.PrivateKey {
+	rsaPrivateKey, err := os.ReadFile("liminal-recovery-key-pair-private-key.pem")
+	if err != nil {
+		log.Println(err)
+		log.Fatal("Error reading rsa private key")
```

Comment:
instead of exiting by using fatal you should return the error from this function. Hence change the function signature as well